instrumentation/github.com/aws/aws-lambda-go/otellambda/example/main.go

```diff
@@ -0,0 +1,114 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-lambda-go/lambda"
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	lambdadetector "go.opentelemetry.io/contrib/detectors/aws/lambda"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"io"
+	"net/http"
+)
+
+func lambda_handler(ctx context.Context)  error {
+	// init aws config
+	cfg, err := awsConfig.LoadDefaultConfig(ctx)
+	if err != nil {
+		panic("configuration error, " + err.Error())
+	}
+
+	// instrument all aws clients
+	otelaws.AppendMiddlewares(&cfg.APIOptions)
+
+	// S3
+	s3Client := s3.NewFromConfig(cfg)
+	input := &s3.ListBucketsInput{}
+	result, err := s3Client.ListBuckets(ctx, input)
+	if err != nil {
+		fmt.Printf("Got an error retrieving buckets, %v", err)
+	}
+
+	fmt.Println("Buckets:")
+	for _, bucket := range result.Buckets {
+		fmt.Println(*bucket.Name + ": " + bucket.CreationDate.Format("2006-01-02 15:04:05 Monday"))
+	}
+
+
+	// HTTP
+	orig := otelhttp.DefaultClient
+	otelhttp.DefaultClient = &http.Client{
+		Transport: otelhttp.NewTransport(
+			http.DefaultTransport,
+			otelhttp.WithTracerProvider(otel.GetTracerProvider()),
+		),
+	}
+	defer func() { otelhttp.DefaultClient = orig }()
+	res, err := otelhttp.Get(ctx, "https://api.github.com/repos/open-telemetry/opentelemetry-go/releases/latest")
```

Comment:
Rather than replacing `otelhttp.DefaultClient` and restoring it later (which I'm not sure is doing much, since it should already have a handle to a delegate to the global tracer provider), perhaps just use `NewTransport()` to create a local `http.Client` instance.

```suggestion
	client := &http.Client{
		Transport: otelhttp.NewTransport(
			http.DefaultTransport,
			otelhttp.WithTracerProvider(otel.GetTracerProvider()),
		),
	}
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://api.github.com/repos/open-telemetry/opentelemetry-go/releases/latest", nil)
	if err != nil {
		//handle err
	}
	res, err := client.Do(req)
```